File: sorting/insertionSort/insertion.c
#include<stdio.h>
int *insertion(int *arr,int n){
    if(n<=1){
        return arr;
    }
    else{
        for(int i=1;i<n;i++){
            int j=i-1;
            int key=arr[i];
            while(j>=0 && key<arr[j]){
                arr[j+1]=arr[j];
                j-=1;
            }
        arr[j+1]=key;
        }
    }
    return arr;
}
int main(){
    int arr[5]={5,3,1,2,4};
    insertion(arr,5);
    for(int i=0;i<5;i++){
        printf("%d ",arr[i]);
    }
}

File: sorting/insertionSort/insertion.py
def insertion(l):
    if len(l)<=1:
        return l
    else :
        for i in range(1,len(l)):
            j=i-1
            key=l[i]
            while(j>=0 and key<l[j]):
                l[j+1]=l[j]
                j-=1
            l[j+1]=key
    return l
l=[5,3,1,2,4]
print(insertion(l))

